### PR TITLE
8342693: Use byte[] as parameter in a FDBigInteger constructor and as field

### PR DESCRIPTION
**Pull request by rgiulietti**
This helps in reducing memory consumption, especially for long inputs.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342693](https://bugs.openjdk.org/browse/JDK-8342693): Use byte[] as parameter in a FDBigInteger constructor and as field (**Enhancement** - P4)

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21614/head:pull/21614` \
`$ git checkout pull/21614`

Update a local copy of the PR: \
`$ git checkout pull/21614` \
`$ git pull https://git.openjdk.org/jdk.git pull/21614/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21614`

View PR using the GUI difftool: \
`$ git pr show -t 21614`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21614.diff">https://git.openjdk.org/jdk/pull/21614.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21614#issuecomment-2426857026)
</details>

### CODE REVIEW

**Comment by rgiulietti** [comment]
This is the first PR of a series aimed at reducing overall CPU and memory consumption in string->floating-point conversions.

File: src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java [hunk 1]
```diff
@@ -1872,7 +1872,7 @@ static ASCIIToBinaryConverter readJavaFormatString( String in ) throws NumberFor
                 }
             }  // look for and process decimal floating-point string
 
-            char[] digits = new char[ len ];
+            byte[] digits = new byte[ len ];
```
Comment (turbanoff) [inline src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java:1875]:
```suggestion
            byte[] digits = new byte[len];
```

**Review by jddarcy — APPROVED** [review]

File: src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java (hunk 1 above)
Comment (rgiulietti) [reply src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java:1875]: @turbanoff I'll apply your suggestions later, in [PR 21840](https://github.com/openjdk/jdk/pull/21840)

**Pull request closed without merging**